### PR TITLE
Introduce Y031: Use class-based syntax for `TypeDict`s wherever possible

### LINKED ISSUE #123: Suggest class-based syntax for a `TypedDict` if-and-only-if it has no invalid field names
Class-based syntax cannot be used if a `TypedDict` has fields that are Python keywords, but should be used in all other circumstances. The `keyword` module could be used to ascertain whether a field is a Python keyword or not.

### PR DESCRIPTION
**Pull request by AlexWaygood**
Closes #123

### CODE REVIEW

**Comment by AlexWaygood** [comment]
typeshed PR here: https://github.com/python/typeshed/pull/6995 (@JelleZijlstra, you only just merged this one! 😉)

**Comment by JelleZijlstra** [comment]
Merged, note there's also a test failure on 3.7. Probably related to ast.Constant which was new in 3.8.

**Comment by AlexWaygood** [comment]
> Merged, note there's also a test failure on 3.7. Probably related to ast.Constant which was new in 3.8.

Yeah. I'm going to file another PR to extend @Akuli's `LegacyNormalizer` PR, then come back to this one, I think.

**Comment by AlexWaygood** [comment]
> Merged, note there's also a test failure on 3.7. Probably related to ast.Constant which was new in 3.8.

`ast.Constant` does exist in 3.8, it just wasn't used in so many situations.

**Comment by AlexWaygood** [comment]
> > Merged, note there's also a test failure on 3.7. Probably related to ast.Constant which was new in 3.8.
> 
> Yeah. I'm going to file another PR to extend @Akuli's `LegacyNormalizer` PR, then come back to this one, I think.

I started working on this, but it ended up being a much more significant change than I realised, so gave up on it for now. Might come back to it later.

File: pyi.py [hunk 1]
```diff
@@ -227,6 +231,43 @@ def visit_Index(self, node: ast.Index) -> ast.expr:
...
+
+def _is_bad_TypedDict(node: ast.Call) -> bool:
+    """Evaluate whether an assignment-based TypedDict should be rewritten using class syntax.
+
+    Return `False` if the TypedDict appears as though it may be invalidly defined;
+    type-checkers will raise an error in that eventuality.
+    """
+
+    args = node.args
+    if len(args) < 2:
```
Comment (Akuli) [inline pyi.py:246]: It's unlikely that anyone will ever add more than 2 args to a TypedDict, but we should probably let type checkers complain if they do:

```suggestion
    if len(args) != 2:
```

Comment (AlexWaygood) [reply pyi.py]: Yes, this makes sense, because `total` is keyword-only. But I should probably add test cases with `total` specified as well.

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**